server/src/services/folders.service.ts

```diff
@@ -106,16 +165,21 @@ export class FoldersService {
         'user_group_permissions',
         'app_group_permissions.group_permission_id = user_group_permissions.group_permission_id'
       )
-      .where('user_group_permissions.user_id = :userId', { userId: user.id })
-      .andWhere('app_group_permissions.read = :value', { value: true })
-      .orWhere(
-        '(viewable_apps.is_public = :value AND viewable_apps.organization_id = :organizationId) ' +
-          'OR viewable_apps.user_id = :userId',
-        {
-          value: true,
-          organizationId: user.organizationId,
-          userId: user.id,
-        }
+      .where('LOWER(viewable_apps.name) like :searchKey', { searchKey: `%${searchKey && searchKey.toLowerCase()}%` })
```

Comment:
Can conditionally search based on the presence of the search key? Doing `like` queries always will be slower.